packages/server/src/modules/zoom/tests/routes.spec.ts

```diff
@@ -1,22 +1,23 @@
-import request from 'supertest';
 import axios from 'axios';
 import jsonwebtoken from 'jsonwebtoken';
 
 import application from '@application';
 import { ZoomRole } from '@chess-tent/models';
+import { TestRequest } from '@types';
 
 beforeAll(() => application.test.start());
 
-const tokenCookie =
-  'token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiMTgwYmQxMDItMjNiNC00MzkxLThhOTUtMzhkYjliN2QxYWJhIn0sImlhdCI6MTY5MTUwMTU2Nn0.UmmBVVMY0Ruxox61CaINh3rtSF73s2V9TD-0nWf2hpw';
+let request: TestRequest;
 
-// rijesit auth
+beforeAll(() => {
+  request = application.test.request;
```

Comment:
Looks interesting. It might work.

Reply:
@aleksmarkovic can you create here a base like this

```
    request = request
      .get('/zoom/authorize')
      .authorize(user)
```

and use it later by just configuring needed difference in test cases.

Reply:
I would leave it like it is because each test has something different, we would be redefining the request then many times. This way we separately build each request for each test and then reset it afterEach.
Except if you have a better PoV?

Reply:
That unwanted effect that you described could be diminished if the request builder was immutable, i.e. if builder methods would return new instance of a builder instead of the same object. By doing that, any change of TestRequest wouldn't be present in other tests in the suite.

Reply:
I'll see what I can do

Reply:
Take a look. If you know a better way of creating a new instance rather than doing it in init would be good.
Now we get a new request instance for every test.